src/swaps/liquid_swapper.rs

```diff
@@ -0,0 +1,992 @@
+            witness: tx_out_witness,
+        };
+        let fee_output: TxOut = TxOut::new_fee(fee, asset_id);
+
+        let mut claim_tx = Transaction {
+            version: 2,
+            lock_time: LockTime::ZERO,
+            input: vec![claim_txin],
+            output: vec![payment_output, fee_output],
+        };
+
+        // If its a cooperative claim, compute the Musig2 Aggregate Signature and use Keypath spending
+        if is_cooperative {
+            let electrum = ElectrumConfig::default_liquid().build_client()?;
+
+            let genesis_blockhash =
+                elements::BlockHash::from_raw_hash(electrum.block_header(0)?.block_hash().into());
+
+            let claim_tx_taproot_hash = SighashCache::new(&claim_tx)
+                .taproot_key_spend_signature_hash(
+                    0,
+                    &Prevouts::All(&[funding_txout]),
+                    SchnorrSighashType::Default,
+                    genesis_blockhash,
+                )
+                .unwrap();
+
+            // Step 6: Generate secret and public nonces
+            let msg = Message::from_digest_slice(claim_tx_taproot_hash.as_byte_array()).unwrap();
+
+            let mut extra_rand = [0u8; 32];
+            OsRng.fill_bytes(&mut extra_rand);
+
+            let (sec_nonce, pub_nonce) = key_agg_cache
+                .nonce_gen(
+                    &secp,
+                    session_id,
+                    self.keypair.public_key(),
+                    msg,
+                    Some(extra_rand),
+                )
+                .unwrap();
+
+            // Step 7: Get boltz's partail sig
+            let claim_tx_hex = serialize(&claim_tx).to_lower_hex_string();
+            let partial_sig_resp = self
+                .api
+                .get_reverse_partial_sig(&reverse_resp.id, &preimage, &pub_nonce, &claim_tx_hex)
+                .unwrap();
+
+            let boltz_nonce =
+                MusigPubNonce::from_slice(&Vec::from_hex(&partial_sig_resp.pub_nonce).unwrap())
+                    .unwrap();
+
+            let boltz_partial_sig = MusigPartialSignature::from_slice(
+                &Vec::from_hex(&partial_sig_resp.partial_signature).unwrap(),
+            )
+            .unwrap();
+
+            // Step 7: Perform
+
+            let agg_nonce = MusigAggNonce::new(&secp, &[boltz_nonce, pub_nonce]);
+
+            let musig_session = MusigSession::new(&secp, &key_agg_cache, agg_nonce, msg);
+
+            let our_partial_sig = musig_session
+                .partial_sign(&secp, sec_nonce, &self.keypair, &key_agg_cache)
+                .unwrap();
+
+            let schnorr_sig = musig_session.partial_sig_agg(&[boltz_partial_sig, our_partial_sig]);
+
+            let final_schnorr_sig = SchnorrSig {
+                sig: schnorr_sig,
+                hash_ty: SchnorrSighashType::Default,
+            };
+
+            // Verify the sigs.
+            let boltz_partial_sig_verify = musig_session.partial_verify(
```

Comment:
You want to verify the partial sig before aggregating them

Reply:
Noted. Will be done in the overriding PR.